ENH: add bdschur, which calls Slycot mb03rd.  Change modal_form to us…

roryyorke opened the pull request:
…e bdschur

The bdschur interface is modelled after the Matlab function of the
same name; it can also optionally sort the modal blocks.

Added tests for bdschur, and modified tests for modal_form.

Closes #318

bnavigator commented on control/canonical.py:
```diff
@@ -149,97 +153,294 @@ def observable_form(xsys):
+
+    Parameters
+    ----------
+    blksizes: size of Schur blocks
+    eigvals: eigenvalues
```
It's a private function, but use numpydoc anyway ( `blksizes : array_like of ...`)?

bnavigator commented on control/canonical.py:
```diff
@@ -149,97 +153,294 @@ def observable_form(xsys):
+
+    Parameters
+    ----------
+    aschur: (n, n) array
```
```suggestion
    aschur: (n, n) array_like
```

bnavigator commented on control/canonical.py:
```diff
@@ -149,97 +153,294 @@ def observable_form(xsys):
-        System in transformed coordinates, with state 'z'
-
+        System in modal canonical form, with state `z`
+    T : matrix
```
matrix?

bnavigator commented on control/tests/canonical_test.py:
```diff
@@ -258,3 +190,250 @@ def test_similarity(self):
+        ([-1,-1,-2,-2,-2], None, [2,3]),
+        ([-1+1j,-1-1j,-2+2j,-2-2j,-2], None, [2,2,1]),
+    ])
+def test_bdschur_ref(eigvals, condmax, blksizes):
```
I guess most of these tests must be marked `@slycotonly`. see Travis CI fails

bnavigator reviewed (commented):
Nice! Some comments about the docstrings and test suite. Can't say much about the algorithms themselves.

roryyorke commented:
Thanks @bnavigator !  I've added the `@slycotonly` decorator, and updated the docs.  I did some minor mods to the other docstrings in `canonical.py` too, and added `bdschur` and `modal_form` to control.rst.

murrayrm commented:
Rebased and changed `_convertToStateSpace` to `_convert_to_statespace`.  Will merge as soon as Travis finishes up.
